service/controller/v21/cloudconfig/worker_template.go

```diff
@@ -156,29 +171,25 @@ func (e *WorkerExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
 		{
 			AssetContent: cloudconfig.DecryptTLSAssetsService,
 			Name:         "decrypt-tls-assets.service",
-			Enable:       false,
-			Command:      "start",
+			Enabled:      true,
```

Comment:
Why was is disabled and now is enabled?

Reply:
the reason for this was that if you reboot already configured machine there was race condition where first run of decrypt fail, I don't remember the details but the solution was disable it and have it run when cloud-config is executed

Reply:
Details are here https://github.com/giantswarm/aws-operator/pull/683

Reply:
I read that and still don't get it. So the unit was run before encrypted cert files were created? Can cloudconfig create units before creating files? Or unit in cloudconfig is basically a file?

Reply:
Try read this https://github.com/giantswarm/aws-operator/pull/683#discussion_r164673864
It is kind of tricky.

Reply:
> The question would still be why is it active now?

Because with ignition it is strict, that files are generated before units. It was not the case with cloudinit.

> Did you make sure the systemd dependencies are met correctly and since ignition is earlier in the boot process the old problem is gone?

I had a lot of tests there, should be fine. Do you see any issues with this?

Reply:
Understanding why it works is important to me. And not only you but also everyone else.

So an explanatory answer could have been:

Ignition is only executed once on first boot. Since it runs before userspace the units generated will be executed in the correct order which is in contrast to cloud-config that was executed much later in the boot process. The important thing here is that the systemd dependency tree needs to be correct so that this unit is executed at the right time which wasn't necessar with the cloud-config. There is no "start" of units anymore with ignition and enabling is necessary since otherwise systemd wouldn't start the unit at all. Neither on first nor on later boots.

Reply:
Got it. Yep, maybe I was not good in explanation. Too many things to follow